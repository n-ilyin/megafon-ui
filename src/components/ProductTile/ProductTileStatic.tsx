import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileStatic.less';
import { IPack } from './types';
import { IProductTileStatic } from './types';
import { productTileStaticProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-static');
class ProductTileStatic extends React.Component<IProductTileStatic> {
    static propTypes = productTileStaticProps;

    render() {
        const { packs } = this.props;

        return (
            <div className={cn('')}>
                {packs.map((item: IPack) =>
                    <div className={cn('constructor-pack')} key={item.title! + item.value! + item.unit!}>
                        {item.isDelim ? item.title : `${item.value} ${item.unit}`}
                    </div>
                )}
            </div>
        );
    }
}

export default ProductTileStatic;
