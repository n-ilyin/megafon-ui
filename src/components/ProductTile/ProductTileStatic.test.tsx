import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileStatic from './ProductTileStatic';

describe('<ProductTileStatic />', () => {
    it('it renders ProductTileStatic', () => {
        const props = {
            packs: [{
                value: 1500,
                unit: 'минут',
            }],
        };
        const wrapper = shallow(<ProductTileStatic {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('it renders ProductTileStatic with delim', () => {
        const props = {
            packs: [{
                title: 'Безлимитный интернет',
                isDelim: true,
            }],
        };
        const wrapper = shallow(<ProductTileStatic {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
