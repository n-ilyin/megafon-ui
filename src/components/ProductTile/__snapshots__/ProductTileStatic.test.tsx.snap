// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileStatic /> it renders ProductTileStatic 1`] = `
<div
  className="mfui-product-tile-static"
>
  <div
    className="mfui-product-tile-static__constructor-pack"
    key="NaNминут"
  >
    1500 минут
  </div>
</div>
`;

exports[`<ProductTileStatic /> it renders ProductTileStatic with delim 1`] = `
<div
  className="mfui-product-tile-static"
>
  <div
    className="mfui-product-tile-static__constructor-pack"
    key="Безлимитный интернетundefinedundefined"
  >
    Безлимитный интернет
  </div>
</div>
`;
